tests/test_manager.py

```diff
@@ -0,0 +1,293 @@
+        agent_alias = self.manager.get_agent_alias(self.agent_name)
+        behaviour = agent_alias.agent.resources.get_behaviour(
+            self.echo_skill_id, "echo"
+        )
+
+        def handler(*args, **kwargs):
+            self.manager.stop_manager()
+
+        self.manager.add_error_callback(handler)
+
+        assert behaviour
+
+        with patch.object(behaviour, "act", side_effect=ValueError("expected")):
+            self.manager.start_all_agents()
+            wait_for_condition(lambda: not self.manager.is_running, timeout=10)
+
+    def test_start_all(self):
+        """Test MultiAgentManager start all agents."""
+        self.test_add_agent()
+        assert self.agent_name in self.manager.list_agents()
+        assert self.agent_name not in self.manager.list_agents(running_only=True)
+        self.manager.start_all_agents()
+        assert self.agent_name in self.manager.list_agents(running_only=True)
+
+        self.manager.start_all_agents()
+
+        with pytest.raises(ValueError, match="is already started!"):
+            self.manager.start_agents(self.manager.list_agents())
+
+        with pytest.raises(ValueError, match="is already started!"):
+            self.manager.start_agent(self.agent_name)
+
+        with pytest.raises(ValueError, match="is not registered!"):
+            self.manager.start_agent("non_exists_agent")
+
+    def test_stop_agent(self):
+        """Test stop agent."""
+        self.test_start_all()
+        wait_for_condition(
+            lambda: self.manager.list_agents(running_only=True), timeout=10
+        )
+        self.manager.stop_all_agents()
+
+        assert not self.manager.list_agents(running_only=True)
+
+        with pytest.raises(ValueError, match=" is not running!"):
+            self.manager.stop_agent(self.agent_name)
+
+        with pytest.raises(ValueError, match=" is not running!"):
+            self.manager.stop_agents([self.agent_name])
+
+    def test_do_no_allow_override_some_fields(self):
+        """Do not allo to override some values in agent config."""
+        self.manager.start_manager()
+
+        self.manager.add_project(self.project_public_id)
+
+        BAD_OVERRIDES = ["skills", "connections", "contracts", "protocols"]
+
+        for bad_override in BAD_OVERRIDES:
+            with pytest.raises(ValueError, match="Do not override any"):
+                self.manager.add_agent(
+                    self.project_public_id,
+                    self.agent_name,
+                    agent_overrides={bad_override: "some value"},
+                )
+
+    @staticmethod
+    def test_invalid_mode():
+        """Test MultiAgentManager fails on invalid mode."""
+        with pytest.raises(ValueError, match="Invalid mode"):
+            MultiAgentManager("test_dir", mode="invalid_mode")
+
+    def test_double_start(self):
+        """Test double MultiAgentManager start."""
+        self.manager.start_manager()
+        self.manager.start_manager()
```

Comment:
maybe could be good to add some asserts in between to check the state of the manager (i.e. idempotency)